v1/format/format.go

```diff
@@ -1207,10 +1207,18 @@ func (w *writer) writeTermParens(parens bool, term *ast.Term, comments []*ast.Co
 	case ast.String:
 		if term.Location.Text[0] == '`' {
 			// To preserve raw strings, we need to output the original text,
-			// not what x.String() would give us.
 			w.write(string(term.Location.Text))
 		} else {
-			w.write(x.String())
+			// x.String() cannot be used by default because it can change the input string "\u0000" to "\x00"
+			// term.Location.Text could contain more than just the string, extract the first string it can find
```

Comment:
Isn't this also an issue for raw strings?

Apparently, we convert them to regular strings:

```rego
x := 1 if {
	false
} else := `foo
bar`
```
->
```rego
x := 1 if {
	false
} else := "foo\nbar"
```

https://play.openpolicyagent.org/p/RMpHXpVG4c

Not expected, but at least not semantically incorrect .. 🤔 
Raw string literals is probably mostly used for constant var declarations anyways ..

Reply:
Is that an issue only when placed in the `else` assignment?
I've never seen that before! And I don't want to see it again, lol

Reply:
This would be an issue for any location raw text that contains more than just the string literal. 
I was thinking `with` might fall under this category, but was wrong: https://play.openpolicyagent.org/p/kQmzCgsBl0